src/PrestaShopBundle/Form/Admin/Extension/MultistoreConfigurationTypeExtension.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Extension;
+
+use PrestaShop\PrestaShop\Core\Domain\Configuration\ShopConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Feature\FeatureInterface;
+use PrestaShop\PrestaShop\Core\Multistore\MultistoreContextCheckerInterface;
+use PrestaShopBundle\Form\Admin\Type\MultistoreConfigurationType;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+class MultistoreConfigurationTypeExtension extends AbstractTypeExtension
+{
+    /**
+     * @var FeatureInterface
+     */
+    private $multistoreFeature;
+
+    /**
+     * @var MultistoreContextCheckerInterface
+     */
+    private $multistoreContext;
+
+    /**
+     * @var ShopConfigurationInterface
+     */
+    private $configuration;
+
+    public function __construct(
+        FeatureInterface $multistoreFeature,
+        MultistoreContextCheckerInterface $multistoreContext,
+        ShopConfigurationInterface $configuration
+    ) {
+        $this->multistoreFeature = $multistoreFeature;
+        $this->multistoreContext = $multistoreContext;
+        $this->configuration = $configuration;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
```

Comment:
Would it be possible to have unit tests for this logic?

I think Symfony has a dedicated PHPUnit Testcase for Twig Extensions that you can use

Reply:
So I extracted the logic in a dedicated service and unit tested it from there.

Reply:
Bravo for the tests!